src/sage/matrix/seymour_decomposition.pyx

```diff
@@ -193,14 +193,67 @@ cdef class SumNode(DecompositionNode):
 
     summands = DecompositionNode._children
 
+    def get_children_matrices(self):
+        if self._children() is None:
+            raise ValueError('Node has no children')
+        return tuple(ch.matrix() for ch in self._children())
+
 
 cdef class OneSumNode(SumNode):
 
-    pass
+    def block_diagonal_repr(self):
+        r"""
```

Comment:
This is a nice idea for a useful method. Some comments:

1) The name won't generalize to `TwoSum` and `ThreeSumNode`; they are still block matrices, but not block diagonal matrices.

2) I would avoid the word `repr` in such names because in Python this means a string representation.